src/_P021_Level.ino

```diff
@@ -182,93 +389,465 @@ boolean Plugin_021(uint8_t function, struct EventStruct *event, String& string)
       String command = parseString(string, 1);
 
       if (equals(command, F("getlevel"))) {
-        string  = toString(P021_TRIGGER_LEVEL);
+        string  = toString(P021_SETPOINT);
         success = true;
       }
       else
       if (equals(command, F("gethysteresis"))) {
-        string  = toString(P021_TRIGGER_HYSTERESIS);
+        string  = toString(P021_HYSTERESIS);
         success = true;
       }
       break;
     }
 
     case PLUGIN_INIT:
     {
-      if (validGpio(CONFIG_PIN1)) {
-        pinMode(CONFIG_PIN1, OUTPUT);
+      if (validGpio(P021_GPIO_RELAY)) {
+        pinMode(P021_GPIO_RELAY, OUTPUT);
       }
       success = true;
       break;
     }
 
+    case PLUGIN_WRITE:
+    {
+      // this case defines code to be executed when the plugin executes an action (command).
+      // Commands can be accessed via rules or via http.
+      // As an example, http://192.168.1.12//control?cmd=dothis
+      // implies that there exists the comamnd "dothis"
+      // Expected commands:
+      // * levelcontrol,remote, [on|off]
+
+      // parse string to extract the command
+      String command = parseString(string, 1); // already converted to lowercase
+      String subcmd  = parseString(string, 2);
+      String value   = parseString(string, 3);
+      String log     = F("P021 write: ");
+
+      if (equals(command, F("levelcontrol"))) {
+        if (equals(subcmd, F("remote"))) {
+          int remote_value = P021_remote[event->TaskIndex];
+          log += F(" levelcontrol");
+
+          if (equals(value, F("on")))
+          {
+            log         += F(" on");
+            remote_value = 1;
+            success      = true;
+          }
+          else if (equals(value, F("off")))
+          {
+            log         += F(" off");
+            remote_value = 0;
+            success      = true;
+          }
+          else
+          {
+            log += F(" **no valid value**");
+          }
+          P021_remote[event->TaskIndex] = remote_value;
+        }
+        else
+        {
+          log += F(" **no valid command**");
+        }
```

Comment:
Not sure if it's needed/useful to tell the user something was wrong, as returning `false` from the `PLUGIN_WRITE` function will already log ` Info   : Command unknown: <your command>`.

Reply:
Thanks. I am quite inexperienced/unsure when using the various options in the event handling. Used if for debugging what my code was doing. Save some memory, loose a tiny bit of information about what part is failing.